src/celeritas/em/distribution/UrbanMscStepLimit.hh

```diff
@@ -114,19 +120,23 @@ class UrbanMscStepLimit
 CELER_FUNCTION
 UrbanMscStepLimit::UrbanMscStepLimit(const UrbanMscRef&       shared,
                                      const ParticleTrackView& particle,
-                                     const PhysicsTrackView&  physics,
+                                     PhysicsTrackView*        physics,
                                      MaterialId               matid,
+                                     bool                     is_first_step,
                                      real_type                safety,
                                      real_type                phys_step)
     : shared_(shared)
+    , physics_(*physics)
     , inc_energy_(value_as<Energy>(particle.energy()))
     , is_positron_(particle.particle_id() == shared.ids.positron)
     , safety_(safety)
     , params_(shared.params)
     , msc_(shared_.msc_data[matid])
-    , helper_(shared, particle, physics)
+    , helper_(shared, particle, physics_)
+    , msc_range_(physics_.msc_range())
+    , on_boundary_(is_first_step || safety_ <= shared.params.safety_limit)
```

Comment:
One consequence of this will be that in some volumes ORANGE may end up recalculating the range at every step since its safety will return 0 when it can't calculate it.

Reply:
Good to know.  We may evaluate differences of the impact between VecGeom and orange.  For our record, there are only two cases for `safety_ <= shared.params.safety_limit` with VecGeom and testem3 geometry; around 9.6% for safety_ = 0 and 4.5% for safety_ = 1.55431e-15 (which is a magic number that I mentioned and introduce this rather arbitrary safety_limit temporarily).

Reply:
Would checking `GeoTrackView::is_on_boundary()` instead be equivalent (and safer for orange)?

Reply:
Definitely support a common definition for that across different geometries and consistent uses thereafter.  For this case, we need to understand the small safety value (1.55431e-15) from VecGeom since there is a visible bad impact on msc results removing the case from the on_boundary condition - which infers that the small safety value is sort of consistent with the on_boundary (equivalent to zero) when VecGeom is used. So, I would like to leave it as it is until we have the better treatment for handling differences between orange and VecGeom.

Reply:
@amandalund Okay, then I need to check whether this small safety value is only from e+ as I have not looked relevant distributions for e-.

Reply:
The small safety value is also from e-, but relatively a little bit less (3.0%).

Reply:
Great!  I did not know that there are cases for `satety_ > msc_params.safety_limit` but a track is still on a boundary. If `GeoTrackView::is_on_boundary()` is a super set of  satety_ <= msc_params.safety_limit, use the is_on_boundary is indeed a good choice as I was also very uncomfortable dealing the arbitrary value for on_boundary condition inside msc (the cut was, in a sense, a conservative choice just to rule out something really close to 1ulp).  I will try to see some of validation plots with replacing the boundary condition by `GeoTrackView::is_on_boundary()` just to confirm that things are better.  Thanks!

Reply:
I confirm that using GeoTrackView::is_on_boundary() indeed improves comparisons with Geant4.  See the same set of electron validation plots (here only with 50 events):
![elec-msc-val-50](https://user-images.githubusercontent.com/26611211/196242301-de23a6f9-63c7-4bac-ba29-4c7193633fe9.png)
![msc-info-elec-50](https://user-images.githubusercontent.com/26611211/196242324-33caaa34-9f2e-4db8-b32b-7a2b49f9b48d.png)

Hard to see the improvement in the first 4 set of plots, but actually the agreement in the first bin is better (i.e., the total number of msc steps is getting closer between celeritas and Geant4): 
| Test cases | Number of electron steps | 
|------------------------| --------------------|
| Celeritas (safety_limit)   | 1184279 |
| Celerita (geo.boundary) | 1246771 |
| Geant4.11.0                    | 1415148 |
    
Will update the code accordingly. Great catch @amandalund !